solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBSolrClient.java

```diff
@@ -480,14 +480,35 @@ private static boolean isTimeExceeded(long timeAllowedNano, long timeOutTime) {
     return timeAllowedNano > 0 && System.nanoTime() > timeOutTime;
   }
 
+  private NamedList<Object> doMakeRequest(Endpoint endpoint, SolrRequest<?> solrRequest)
+      throws SolrServerException, IOException {
+    final var solrClient = getClient(endpoint);
+    return doMakeRequest(solrClient, endpoint.getBaseUrl(), endpoint.getCore(), solrRequest);
+  }
+
+  // TODO This special casing can be removed if either: (1) SOLR-16367 is completed, or (2)
+  // LBHttp2SolrClient.getClient() is modified to return a client already pointed at the correct URL
+  private NamedList<Object> doMakeRequest(
+      SolrClient solrClient, String baseUrl, String collection, SolrRequest<?> solrRequest)
+      throws SolrServerException, IOException {
+    // Some implementations of LBSolrClient.getClient(...) return a Http2SolrClient that may not be
+    // pointed at the desired URL (or any URL for that matter).  We special case that here to ensure
+    // the appropriate URL is provided.
+    if (solrClient instanceof Http2SolrClient) {
+      final var httpSolrClient = (Http2SolrClient) solrClient;
+      return httpSolrClient.requestWithBaseUrl(baseUrl, (c) -> c.request(solrRequest, collection));
+    }
+
+    return solrClient.request(solrRequest, collection);
```

Comment:
then how is baseUrl applied?   (after reading below) okay I see.  A comment directly above this line saying "assume solrClient is pointed at baseUrl" would be very helpful.

Note there could be a JDK client specific issue as as @jdyer1 is presently working on JDK client specific support.

Reply:
> A comment directly above this line ...

Done

> Note there could be a JDK client specific issue as as @jdyer1 is presently working on JDK client specific support.

Yeah, my first thought on seeing that PR was for the ramifications here.  Will take a look at his PR to make sure it plays nice with this code.

All of this logic would be much cleaner if all implementations of `LBSolrClient.getClient(baseUrl)` returns a client that was "ready to go".  The apache impl returns a HttpSolrClient already pointed at the right URL, but the Jetty impl doesn't.

Reply:
Agreed, but IMO that probably deserves its own ticket.

Switching the Jetty LB client to work this way would probably let us reuse/share some of the client-management code from the Apache LB client...which is great!...but it'd also turn things into a slightly larger refactor than I want to tackle here.

If we're agreed on this approach I can create a ticket for that work and update the TODO comment here to say essentially: "rip this out when tackling SOLR-12345"?

Reply:
I agree it's definitely worth its own ticket!

Reply:
Great - just created SOLR-17541, and updated the TODO comment around this method accordingly